src/types/orchestrationTypes.ts

```diff
@@ -1,13 +1,18 @@
 import { FunctionOptions, FunctionTrigger, InvocationContext } from "@azure/functions";
-import { TokenSource } from ".";
+import { Task, TokenSource } from ".";
 import { DurableOrchestrationContext } from "../classes";
 
 /**
  * Type of a Generator that can be registered as an orchestration
+ * @param T the return value of the orchestration
  */
-export type OrchestrationHandler = (
+export type OrchestrationHandler<T = any> = (
     context: OrchestrationContext
-) => Generator<unknown, unknown, any>;
+) => Generator<
+    Task, // orchestrations yield Task types
```

Comment:
By contrast to my above comment, changing `unknown` to `Task` feels more obviously beneficial and I would support getting this part of the PR merged